frontend/src/components/ScheduledInquiries/AddScheduledInquiry.tsx

```diff
@@ -1,32 +1,203 @@
-import { Button } from "@chakra-ui/react"
+import { Button, Flex, Switch } from "@chakra-ui/react"
+import { useMutation } from "@tanstack/react-query"
 import { useState } from "react"
-import { type InquiryPublic, ScheduledInquiriesService } from "../../client"
+import { FiChevronDown, FiChevronUp, FiEdit2, FiTrash } from "react-icons/fi"
+import {
+  type ApiError,
+  InquiriesService,
+  type InquiryPublic,
+  ScheduledInquiriesService,
+} from "../../client"
+import UpdateInquiry from "../../components/Inquiries/UpdateInquiry"
+import useCustomToast from "../../hooks/useCustomToast"
+import { handleError } from "../../utils/showToastOnError.ts"
 import FormModal from "../Common/FormModal.tsx"
 
 type AddScheduledInquiryProps = {
   inquiry: InquiryPublic
+  refetchInquiries: () => Promise<void>
 }
-const AddScheduledInquiry = ({ inquiry }: AddScheduledInquiryProps) => {
+const AddScheduledInquiry = ({
+  inquiry,
+  refetchInquiries,
+}: AddScheduledInquiryProps) => {
   const [isModalOpen, setModalOpen] = useState(false)
+  const [isUpdateModalOpen, setUpdateModalOpen] = useState(false)
+
+  const showToast = useCustomToast()
 
   const addToScheduledInquiries = async () => {
-    return ScheduledInquiriesService.addToSchedule({
+    const addToSchedule = ScheduledInquiriesService.addToSchedule({
       requestBody: {
         inquiry_id: inquiry.id,
       },
     })
+    await addToSchedule
+    await refetchInquiries()
+    return addToSchedule
   }
 
+  const deleteInquiryMutation = useMutation({
+    mutationFn: () => {
+      return InquiriesService.deleteInquiry({
+        requestBody: {
+          id: inquiry.id,
+          text: inquiry.text,
+          theme_id: inquiry.theme_id,
+        },
+      })
+    },
+    onSuccess: async () => {
+      await refetchInquiries()
+      showToast("Success!", "Inquiry deleted", "success")
+    },
+    onError: (err: ApiError) => {
+      handleError(err, showToast)
+    },
+    onSettled: () => {
+      // if (queryKeyToInvalidate) {
```

Comment:
should the commented out code be removed?

Reply:
Yes, it should be. I did this in a couple other spots that I will address too.